pkg/paillier/secret.go

```diff
@@ -28,21 +29,21 @@ var (
 type SecretKey struct {
 	*PublicKey
 	// p, q such that N = p⋅q
-	p, q *big.Int
+	p, q *safenum.Nat
 	// phi = ϕ = (p-1)(q-1)
-	phi *big.Int
+	phi *safenum.Nat
 	// phiInv = ϕ⁻¹ mod N
-	phiInv *big.Int
+	phiInv *safenum.Nat
 }
 
 // P returns the first of the two factors composing this key.
 func (sk *SecretKey) P() *big.Int {
```

Comment:
For later, we should definitely return a safe Int here !

Reply:
Yeah, once I knock off the final zk proof, my plan is to modify the different APIs to return safenum stuff judiciously, and then glue it all together.

Paillier is probably only going to accept safenum stuff, because the values you're encrypting are homomorphically mixing in are secret, most often.